include/boost/geometry/extensions/random/strategies/agnostic/uniform_convex_hull_rejection.hpp

```diff
@@ -0,0 +1,82 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2019 Tinko Bartels, Berlin, Germany.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_EXTENSIONS_RANDOM_STRATEGIES_AGNOSTIC_UNIFORM_CONVEX_HULL_REJECTION_HPP
+#define BOOST_GEOMETRY_EXTENSIONS_RANDOM_STRATEGIES_AGNOSTIC_UNIFORM_CONVEX_HULL_REJECTION_HPP
+
+#include <random>
+#include <vector>
+
+#include <boost/geometry/algorithms/equals.hpp>
+#include <boost/geometry/algorithms/transform.hpp>
+#include <boost/geometry/algorithms/within.hpp>
+#include <boost/geometry/geometries/polygon.hpp>
+#include <boost/geometry/algorithms/convex_hull.hpp>
+
+#include <boost/geometry/extensions/random/strategies/agnostic/uniform_convex_fan.hpp>
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace uniform_point_distribution {
+
+// The following strategy is suitable for geometries for which
+// a Triangle sampling strategy can be provided
+// a SideStrategy that compues the triangle area can be provided.
+template
+<
+    typename Point,
+    typename DomainGeometry,
+    typename TriangleStrategy,
+    typename SideStrategy   //Actually, we need a triangle area strategy here.
+>
+struct uniform_convex_hull_rejection
+{
+private:
+    typedef typename point_type<DomainGeometry>::type domain_point_type;
+    typedef boost::geometry::model::ring<domain_point_type> ring;
+    ring hull;
+    uniform_convex_fan<Point, ring, TriangleStrategy, SideStrategy>
+        m_strategy;
+public:
+    uniform_convex_hull_rejection(DomainGeometry const& g) : m_strategy(hull)
+    {
+        boost::geometry::convex_hull(g, hull);
+        m_strategy =
+            uniform_convex_fan
+                <
+                    Point,
+                    ring,
+                    TriangleStrategy,
+                    SideStrategy
+                >(hull);
+    }
+    bool equals(DomainGeometry const& l_domain,
+                DomainGeometry const& r_domain,
+                uniform_convex_hull_rejection const& r_strategy) const
+    {
+        return boost::geometry::equals(l_domain, r_domain)
+            && m_strategy.equals(l_domain, r_domain, r_strategy.m_strategy);
+    }
+    template<typename Gen>
+    Point apply(Gen& g, DomainGeometry const& d)
+    {
+        Point p;
+        do{
```

Comment:
By the implementation of uniform_point_distribution which is the user of all those strategies, hull will always be the convex hull of d, hence their intersection must be non-empty if d has a non-empty interior. I can see that this is problematic since the strategy in itself is an exposed interface that could be misused by passing different geometries to constructor and apply-call. I will defer that issue since it may be rendered moot by future refactoring because of the question of what belongs into strategies.

If d (as an areal geometry) has an empty interior (area is zero), I consider it to be invalid input since I am sampling from the interior with the areal types. I understand that this will need to be documented.